safenode/src/network/event.rs

```diff
@@ -167,8 +167,24 @@ impl SwarmDriver {
                     }
                 }
             }
-            SwarmEvent::ConnectionClosed { peer_id, .. } => {
-                info!("Peer {} has gone offline", peer_id);
+            SwarmEvent::ConnectionClosed {
+                peer_id, endpoint, ..
+            } => {
+                // The connection closed due to no msg flow show as `endpoint::receiver`
```

Comment:
Is there nothing we need to do with this?
A ping or so perhaps?

Reply:
libp2p itself will carry out a dail attempt by itself.
under the current config, seems we don't need to do anything so far.